src/client/Microsoft.Identity.Client/MsalServiceExceptionFactory.cs

```diff
@@ -61,7 +64,11 @@ internal static MsalServiceException FromHttpResponse(
                     innerException);
             }
 
-            ex ??= new MsalServiceException(errorCode, errorMessage, innerException);
+            ex ??= new MsalServiceException(errorCode,
+                httpResponse.StatusCode == HttpStatusCode.NotFound
+                        ? errorMessage + " Authority used: " + context.ApiEvent.TokenEndpoint.Split('?')[0]
```

Comment:
```suggestion
                        ? errorMessage + " TokenEndpoint used: " + context.ApiEvent.TokenEndpoint.Split('?')[0]
```

Reply:
This is intended to be the authority, not the token endpoint, I will update the value